### PR TITLE
"make samples" should build all samples

### PR DESCRIPTION
**Pull request by cabelitos**

### CODE REVIEW

**Comment by ceolin** [comment]
lgtm

File: src/samples/flow/am2315/Kconfig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+config FLOW_AM2315_SAMPLE
+	bool
+	depends on FLOW_FBP_GENERATOR_SAMPLES && FLOW_NODE_TYPE_AM2315
```
Comment (edersondisouza) [inline src/samples/flow/am2315/Kconfig]: Don't you have to depend on all node types used on the sample? I know that some samples do that and other don't, but we need consistency.

File: src/samples/flow/http-oauth/Makefile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+sample-$(FLOW_HTTP_OATH_TWITTER_SAMPLE) += oauth_twitter
```
Comment (edersondisouza) [inline src/samples/flow/http-oauth/Makefile]: s/OATH/OAUTH/

File: src/samples/flow/http-oauth/Kconfig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+config FLOW_HTTP_OATH_TWITTER_SAMPLE
```
Comment (edersondisouza) [inline src/samples/flow/http-oauth/Kconfig]: Ditto-¹

File: src/modules/flow-metatype/js/js_code_start.h [hunk 4]
```diff
@@ -720,6 +721,9 @@
     "}\n" \
     "static int\n" \
     "js_metatype_simple_port_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,\n" \
+    "    const struct sol_flow_packet *packet) SOL_ATTR_UNUSED;\n" \
+    "static int\n" \
+    "js_metatype_simple_port_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,\n" \
```
Comment (edersondisouza) [inline src/modules/flow-metatype/js/js_code_start.h]: I didn't understand this. It appears that you duplicated `js_metatype_simple_port_process`, one with the attribute and other without it. Am I missing something?

File: src/modules/flow-metatype/js/js_code_start.h [hunk 5]
```diff
@@ -733,6 +737,8 @@
     "    duk_pop_n(*duk_ctx, 3);\n" \
     "    return r;\n" \
     "}\n" \
+    "static int js_metatype_composed_port_process(struct sol_flow_node *node, void *data,\n" \
+    "    uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet) SOL_ATTR_UNUSED;\n" \
```
Comment (edersondisouza) [inline src/modules/flow-metatype/js/js_code_start.h]: Ditto.

File: src/samples/flow/am2315/Kconfig (hunk 1 above)
Comment (cabelitos) [inline src/samples/flow/am2315/Kconfig]: I realized that. I plan to create another PR normalizing this issue

File: src/modules/flow-metatype/js/js_code_start.h (hunk 4 above)
Comment (cabelitos) [inline src/modules/flow-metatype/js/js_code_start.h]: I need to set SOL_ATTR_UNUSED at the function prototype. Otherwise gcc complains

**Comment by cabelitos** [comment]
merged

**Pull request closed without merging**